### PR TITLE
Verify has events

### PR DESCRIPTION
**Pull request by talosdennis**
I have added the hasEvents property to the Category entity. In the Events service, whenever you make an edit/create/delete operation on an event, we look for its corresponding category and update its hasEvents field.

### CODE REVIEW

File: api/DTOs/CategoryDto.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ public class CategoryDto
 		public string Icon { get; set; }
 		[Required]
 		public string DefaultBackground { get; set; }
+		[Required]
```
Comment (larisaChh) [inline api/DTOs/CategoryDto.cs]: Why is this required?

File: api/Service/EventService.cs [hunk 2]
```diff
@@ -48,16 +49,31 @@ public async Task<Event> CreateEvent(EventDto eventDto)
...
 		{
 			var myEvent = await _context.Events.FindAsync(eventID);
+
 			if (myEvent == null)
 			{
 				return false;
 			}
+
+			var eventsWithSameCategory = await _context.Events.Where(ev => ev.CategoryId == myEvent.CategoryId).ToListAsync();
+			if (eventsWithSameCategory.Count == 1)
```
Comment (Mihai-Sarmaru-Yonder) [inline api/Service/EventService.cs]: You can use SingleOrDefaultAsync() instead of Where and check if it's not null.

File: api/Service/EventService.cs [hunk 3]
```diff
@@ -89,6 +105,17 @@ public async Task<Event> PutEvent(Event eventToPut)
 			{
 				return null;
 			}
+			if (eventToPut.CategoryId != myEvent.CategoryId)
+			{
+				var eventsWithSameCategory = await _context.Events.Where(ev => ev.CategoryId == myEvent.CategoryId).ToListAsync();
+				if (eventsWithSameCategory.Count == 1)
```
Comment (Mihai-Sarmaru-Yonder) [inline api/Service/EventService.cs]: Same here. And since it's duplicate code, maybe extracting the logic to a private method returning bool could be a better approach.

File: api/Service/EventService.cs (hunk 2 above)
Comment (talosdennis) [reply api/Service/EventService.cs:70]: Is it okay if I use FirstOrDefaultAsync instead of SingleOrDefaultAsync() ?

File: api/Service/EventService.cs [hunk 4]
```diff
@@ -48,16 +49,30 @@ public async Task<Event> CreateEvent(EventDto eventDto)
 				Tags = eventDto.Tags,
 				BackgroundId = await _pictureService.CreatePictureByContent(eventDto.BackgroundImage)
 			};
+
+			var updateCategory = await _categoryService.GetCategory(eventDto.CategoryId);
+			updateCategory.HasEvents = true;
+			await _categoryService.PutCategory(eventDto.CategoryId, updateCategory);
+
```
Comment (larisaChh) [inline api/Service/EventService.cs]: this should be a new function 

updateCategoryHasEvents(bool hasEvents) {
             var updateCategory = await _categoryService.GetCategory(eventDto.CategoryId);
			updateCategory.HasEvents = hasEvents;
			await _categoryService.PutCategory(eventDto.CategoryId, updateCategory);
}

Comment (larisaChh) [inline api/Service/EventService.cs]: this should be a new function 

updateCategoryHasEvents(bool hasEvents) {
             var updateCategory = await _categoryService.GetCategory(eventDto.CategoryId);
			updateCategory.HasEvents = hasEvents;
			await _categoryService.PutCategory(eventDto.CategoryId, updateCategory);
}

Comment (larisaChh) [reply api/Service/EventService.cs]: you can use it below also on delete event and put event

Comment (larisaChh) [reply api/Service/EventService.cs]: you can use it below also on delete event and put event

File: api/Service/EventService.cs [hunk 5]
```diff
@@ -181,5 +208,17 @@ public async Task<IList<Event>> GetFilteredEvents(FiltersDto filtersDto)
 								.ToListAsync();
 			return futureJoinedEvents;
 		}
+
+		private async Task<bool> VerifyCategory(Event myEvent)
+		{
+			//to-do:
+			//make it work with SingleOrDefaultAsync();
+			var eventsWithSameCategory = await _context.Events.Where(ev => ev.CategoryId == myEvent.CategoryId).ToListAsync();
+			if (eventsWithSameCategory.Count == 1)
```
Comment (larisaChh) [inline api/Service/EventService.cs]: return eventsWithSameCategory.Count == 1
